libraries/ESP_NOW/examples/ESP_NOW_Serial/ESP_NOW_Serial.ino

```diff
@@ -0,0 +1,107 @@
+/*
+    ESP-NOW Serial Example - Unicast transmission
+    Lucas Saavedra Vaz - 2024
+    Send data between two ESP32s using the ESP-NOW protocol in one-to-one (unicast) configuration.
+    Note that different MAC addresses are used for different interfaces.
+    To properly visualize the data being sent, set the line ending in the Serial Monitor to "Both NL & CR".
+*/
+
+#include "ESP32_NOW_Serial.h"
+#include "MacAddress.h"
+#include "WiFi.h"
+
+// 0: AP mode, 1: Station mode
+#define ESPNOW_WIFI_MODE_STATION 0
+
+// Channel to be used by the ESP-NOW protocol from 1 to 13
+#define ESPNOW_WIFI_CHANNEL 1
+
+#if ESPNOW_WIFI_MODE_STATION // ESP-NOW using WiFi Station mode
+#define ESPNOW_WIFI_IF   WIFI_IF_STA
+#define GET_IF_MAC       WiFi.macAddress
+
+// Station mode MAC addresses of the devices in the network
+// Device 1 - F4:12:FA:42:B6:E8
+const MacAddress mac1({0xF4, 0x12, 0xFA, 0x42, 0xB6, 0xE8});
+// Device 2 - F4:12:FA:40:64:4C
+const MacAddress mac2({0xF4, 0x12, 0xFA, 0x40, 0x64, 0x4C});
+#else // ESP-NOW using WiFi AP mode
+#define ESPNOW_WIFI_IF   WIFI_IF_AP
+#define GET_IF_MAC       WiFi.softAPmacAddress
+
+// AP mode MAC addresses of the devices in the network
+// Device 1 - F6:12:FA:42:B6:E8
+const MacAddress mac1({0xF6, 0x12, 0xFA, 0x42, 0xB6, 0xE8});
+// Device 2 - F6:12:FA:40:64:4C
+const MacAddress mac2({0xF6, 0x12, 0xFA, 0x40, 0x64, 0x4C});
+#endif
+
+ESP_NOW_Serial_Class *esp_now_serial;
+
+void setup() {
+    MacAddress current_mac;
+
+    Serial.begin(115200);
+
+    Serial.print("WiFi Mode: ");
+
+    #if ESPNOW_WIFI_MODE_STATION
+    Serial.println("STA");
+    WiFi.mode(ESPNOW_WIFI_MODE);
+    esp_wifi_set_channel(ESPNOW_WIFI_CHANNEL, WIFI_SECOND_CHAN_NONE);
```

Comment:
We should have API for this.

Reply:
In the ESP-NOW?

Reply:
https://github.com/espressif/arduino-esp32/pull/9405